kernel/mempool.c

```diff
@@ -46,9 +62,13 @@ void k_free(void *ptr)
 K_HEAP_DEFINE(_system_heap, CONFIG_HEAP_MEM_POOL_SIZE);
 #define _SYSTEM_HEAP (&_system_heap)
 
-void *k_malloc(size_t size)
+void *k_aligned_alloc(size_t align, size_t size)
 {
-	return z_heap_malloc(_SYSTEM_HEAP, size);
+	__ASSERT(align / sizeof(void *) >= 1
+		&& (align % sizeof(void *)) == 0,
+		"align must be a multiple of sizeof(void *)");
```

Comment:
Eventually, `sys_heap_aligned_alloc()` will be called to allocate memory, and on master, it has an assertion that align must be power of 2.

Reply:
Yes. Originally, I had the PoT check there as well, but decided to drop it in `k_aligned_alloc()` because it would eventually be met. I think not seeing it here is more confusing than not, so I'll put it back.

Reply:
hm... with the assertion below, is this still necessary? Besides, it is possible for this to be called with alignment of 1. For example, some `#define` is used to specify alignment due to kconfigs in some header files (which could be 1, as in "doesn't really matter"). So at function call site, there is no need to `#if/#else` to force usage of `k_malloc()` when alignment doesn't really matter.